### PR TITLE
Add tests to `test_responses`

### PR DESCRIPTION
**Pull request by Orenoid**
# Summary

Add tests for some of the uncovered branches in `starlette.responses`, related to [this issue](https://github.com/encode/starlette/issues/2452)

# Checklist

- [x] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly. (It seems no documentation needs update)

### CODE REVIEW

File: tests/test_responses.py [hunk 1]
```diff
@@ -381,6 +396,29 @@ async def app(scope: Scope, receive: Receive, send: Send) -> None:
...
+            httponly=False,
+            samesite=None,
+        )
+        await response(scope, receive, send)
+
+    client = test_client_factory(app)
+    response = client.get("/")
+    assert response.headers["set-cookie"] == "mycookie=myvalue"
+
+
```
Comment (Orenoid) [inline tests/test_responses.py]: https://github.com/encode/starlette/blob/master/starlette/responses.py#L105-L120
This is testing the circumstance where all those conditions are `False`.

File: tests/test_responses.py [hunk 2]
```diff
@@ -287,6 +287,21 @@ async def send(message: Message) -> None:
...
+    path = os.path.join(tmpdir, "xyz")
+    content = b"<file content>"
+    with open(path, "wb") as file:
+        file.write(content)
+    # By default, FileResponse will determine the `media_type` based on
+    # the filename or path, unless a specific `media_type` is provided.
+    app = FileResponse(path=path, filename="example.png", media_type="image/jpeg")
+    client: TestClient = test_client_factory(app)
+    response = client.get("/")
+    assert response.headers["content-type"] == "image/jpeg"
```
Comment (Orenoid) [inline tests/test_responses.py]: https://github.com/encode/starlette/blob/master/starlette/responses.py#L294-L295
Testing the uncovered branch of this `if` statement.

**Review by Kludex — APPROVED** [review]
Thanks! 🙏

**Comment by Orenoid** [comment]
@Kludex  Thanks for the approval. I noticed that you updated some logic and variable names. Do we have relevant coding standards? If so, I can follow them while working on the rest of tests in the future.
And one more thing, would you prefer me to add the rest of the tests in separate PRs, with one PR per module, or should I add them with as few PRs as possible?

**Comment by Kludex** [comment]
> Do we have relevant coding standards?

We do have a page in our docs, but for the changes I did in this PR, no. I just updated the code to use `tmp_path` instead of `tmpdir`. You didn't do anything wrong.

> And one more thing, would you prefer me to add the rest of the tests in separate PRs, with one PR per module, or should I add them with as few PRs as possible?

The way you did here is perfect! A couple of tests per PR is good for me.

**Pull request merged**